mirascope/logfire/logfire.py

```diff
@@ -379,19 +379,57 @@ def with_logfire(cls):
         setattr(cls, "add", mirascope_logfire_span(cls.add))
     if get_parent_class_name(cls, "OpenAI"):
         if hasattr(cls, "call_params"):
-            cls.call_params.logfire = logfire.instrument_openai
+            setattr(
```

Comment:
we likely also want to do this with `weave`?

Reply:
done